### PR TITLE
Channels

### PR DESCRIPTION
**Pull request by strongliang**
add a command to admin.js to gather lists of services seen on hyperbahn

### CODE REVIEW

File: test/admin/channels.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}, function t(cluster, assert) {
+    Admin.exec('channels query', {
+        hosts: cluster.hostPortList
+    }, function onQuery(err, arr) {
+        assert.ifError(err);
+        assert.equal(arr.length, 4);
+        for (var i = 0; i < arr.length; i++) {
+            var result = arr[i];
+            assert.ok(result.channels);
+            assert.notEqual(result.channels, 0);
```
Comment (Raynos) [inline test/admin/channels.js]: channels.length ?

File: test/admin/channels.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    size: 4
+}, function t(cluster, assert) {
+    Admin.exec('channels query', {
+        hosts: cluster.hostPortList
+    }, function onQuery(err, arr) {
+        assert.ifError(err);
+        assert.equal(arr.length, 4);
+        for (var i = 0; i < arr.length; i++) {
+            var result = arr[i];
+            assert.ok(result.channels);
```
Comment (Raynos) [inline test/admin/channels.js]: can we assert that `ringpop` and `autobahn` are always in the results ?

File: test/admin/channels.js (hunk 1 above)
Comment (strongliang) [inline test/admin/channels.js]: actually it's an object.

File: test/admin/channels.js (hunk 2 above)
Comment (strongliang) [inline test/admin/channels.js]: sure

**Comment by Raynos** [comment]
Can we add a json output mode ?

Outputting line seperated txt is human readable but not machine readable. Adding a json output mode allows us to use jq to analyze data or even use grep.

**Comment by strongliang** [comment]
```
  $$ node admin.js channels query -j | grep -v finished | jq .
    {
      "host": "127.0.0.1:21300",
      "channels": [
        "autobahn",
        "ringpop",
        "hyperbahn"
      ]
    }
    {
      "host": "127.0.0.1:21301",
      "channels": [
        "autobahn",
        "ringpop",
        "hyperbahn"
      ]
    }
```

**Comment by Raynos** [comment]
@strongliang #noice 

this looks good to me.

Let's get another reviewer ( @jcorbin @rf ) on monday.

File: bin/admin.js [hunk 3]
```diff
@@ -203,6 +211,7 @@ function send(endpoint, body, query, cb) {
     }
 
     function done(err, results) {
+        // console.log(results);
```
Comment (jcorbin) [inline bin/admin.js]: obsolete debug log?

**Comment by jcorbin** [comment]
:ok:

**Comment by Raynos** [comment]
do it. #mergeit

**Comment by Raynos** [comment]
This is shipped.

**Pull request merged**